shared/fs/row/drop-target.desktop.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import * as React from 'react'
+import {Box, Icon} from '../../common-adapters'
+import * as Styles from '../../styles'
+
+type State = {|
+  showDropOverlay: boolean,
+|}
+
+const DropOverlay = ({onDragLeave, onDrop}) => (
+  <Box style={styles.dropOverlayStyle} onDragLeave={onDragLeave} onDrop={onDrop}>
+    <Icon type="icon-dropping-file-48" />
+  </Box>
+)
+
+class DropTarget extends React.PureComponent<any, State> {
+  state = {showDropOverlay: false}
+
+  _onDrop = e => {
+    if (!this._validDrag(e)) {
+      return
+    }
+    const fileList = e.dataTransfer.files
+    const paths = fileList.length ? Array.prototype.map.call(fileList, f => f.path) : []
+    this.props.onAttach(paths)
+    this.setState({showDropOverlay: false})
+  }
+
+  _validDrag = e => Array.prototype.map.call(e.dataTransfer.types, t => t).includes('Files')
+
+  _onDragOver = e => {
+    if (this._validDrag(e)) {
+      e.dataTransfer.dropEffect = 'copy'
+      this.setState({showDropOverlay: true})
+    } else {
+      e.dataTransfer.dropEffect = 'none'
+    }
+  }
+
+  _onDragLeave = e => {
+    this.setState({showDropOverlay: false})
+  }
+
+  render() {
+    const {children, onAttach, ...otherProps} = this.props
+    return (
+      <Box style={styles.containerStyle} onDragOver={onAttach ? this._onDragOver : null} {...otherProps}>
+        {children}
+        {this.state.showDropOverlay && <DropOverlay onDragLeave={this._onDragLeave} onDrop={this._onDrop} />}
+      </Box>
```

Comment:
I wonder if it's possible to have the `DropOverlay` along side with `Rows`, instead of having them both as children in here.

Reply:
Not sure that is possible. Now it only has the extra box if necessary.

Reply:
Could probably add a `<DropOverlay />` [here](https://github.com/keybase/client/blob/aa76cc7a1c4e06e80210e514a5650cb59b1978ab/shared/fs/folder/index.js#L32) above `<Rows />`? But I don't feel strongly about sibling vs children, as long as the Rows component doesn't need to know about drag/drop.

Reply:
But there still needs to be a Box with onDragOver to trigger the DropOverlay (and Box2 does not work unless extended).

Reply:
But the DropTarget can be moved fs/folder/index.js of course.